unlock-app/src/hooks/useCheckoutCommunication.ts

```diff
@@ -195,11 +199,30 @@ export const useCheckoutCommunication = () => {
         })
       },
       sendAsync: (request: MethodCall, callback) => {
+        if (!request.id) {
+          request.id = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER)
+        }
         waitingMethodCalls[request.id] = (error: any, response: any) => {
           callback(error, response)
         }
         emitMethodCall(request)
       },
+      request: async (request: MethodCall) => {
+        if (!request.id) {
+          // Assigning an id because they may be returned in a different order
+          request.id = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER)
```

Comment:
Use crypto uuid if it always needs to be unique.

Reply:
It does not really need to be always unique, just unique within a short time frame (between the request and response!)